.github/workflows/build-multiarch-manifest.yml

```diff
@@ -0,0 +1,80 @@
+name: Build multi-arch docker image
+
+on:
+  workflow_call:
+    inputs:
+      ARCHES:
+        required: true
+        type: string
+      CUDA_VER:
+        required: true
+        type: string
+      LINUX_VER:
+        required: true
+        type: string
+      PYTHON_VER:
+        required: true
+        type: string
+      IMAGE_REPO:
+        required: true
+        type: string
+      BUILD_TYPE:
+        required: true
+        type: string
+
+jobs:
+  get-prefix:
+    runs-on: ubuntu-latest
+    outputs:
+        FULL_PREFIX: ${{ steps.get-prefix.outputs.FULL_PREFIX }}
+        USED_REPO: ${{ steps.get-prefix.outputs.PREFIX }}
+        TAG_PREFIX: ${{ steps.get-prefix.outputs.TAG_PREFIX }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Compute Prefix
+        id: get-prefix
+        uses: ./.github/actions/compute-prefix
+        with:
+          ORIGINAL_PREFIX: rapidsai/${{ inputs.IMAGE_REPO }}
+          IMAGE_REPOSITORY: ${{ inputs.IMAGE_REPO }}
+          BUILD_TYPE: ${{ inputs.BUILD_TYPE }}
```

Comment:
the computed prefix doesn't change per-workflow, right?

if so, this job should be hoisted to the top-level `.github/workflows/build-and-publish-images.yml` workflow so that it's not run in every matrix job. that's redundant and not a good way to spend CI resources.

you should consider renaming the `compute-matrix` job in `.github/workflows/build-and-publish-images.yml` to `compute-properties` and compute all of the workflow constants there.

the properties can then be passed to `build-image` and `build-multiarch-manifest` as inputs so we're not computing the same prefix a million times.

Reply:
> the computed prefix doesn't change per-workflow, right?

I want to be sure I understand the question correctly. The computed prefix is formed from the `IMAGE_REPO` entries stored in `axis.yaml`. That value is not the same for all items in the matrix; and if we're pushing to staging (for PR workflows), the prefix _may_ need to be computed for each variant of the matrix.